sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1406,8 +1406,16 @@ public boolean apply(@Nonnull String input) {
                       klass, entry.getKey(), closestMatches));
           }
         }
-
+        // We need to pull the method directly from klass, to retain generic information.
         Method method = propertyNamesToGetters.get(entry.getKey());
+        String methodName = method.getName();
+        try {
+          method = klass.getMethod(methodName);
+        } catch (NoSuchMethodException e) {
+          // If the method is not found, fall back to just using the registered value.  This
```

Comment:
I updated this to use the same logic as allInterfaceMethods, but must have misunderstood something, because it breaks many tests.  Could you provide more detail, please?

    SortedSet<Method> methods =
        FluentIterable
            .from(ReflectHelpers.getClosureOfMethodsOnInterface(beanClass))
            .filter(NOT_SYNTHETIC_PREDICATE)
            .toSortedSet(MethodComparator.INSTANCE);